x-pack/plugins/reporting/public/plugin.tsx

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import * as Rx from 'rxjs';
+import { catchError, filter, map, mergeMap, takeUntil } from 'rxjs/operators';
+import { i18n } from '@kbn/i18n';
+import {
+  CoreSetup,
+  CoreStart,
+  Plugin,
+  PluginInitializerContext,
+} from '../../../../src/core/public';
+import {
+  JOB_COMPLETION_NOTIFICATIONS_POLLER_CONFIG,
+  JOB_COMPLETION_NOTIFICATIONS_SESSION_KEY,
+} from '../constants';
+import { JobId, JobStatusBuckets, NotificationsService } from '../index.d';
+import { getGeneralErrorToast } from './components';
+import { ReportingNotifierStreamHandler as StreamHandler } from './lib/stream_handler';
+
+const {
+  jobCompletionNotifier: { interval: JOBS_REFRESH_INTERVAL },
+} = JOB_COMPLETION_NOTIFICATIONS_POLLER_CONFIG;
+
+function getStored(): JobId[] {
+  const sessionValue = sessionStorage.getItem(JOB_COMPLETION_NOTIFICATIONS_SESSION_KEY);
+  return sessionValue ? JSON.parse(sessionValue) : [];
+}
+
+function handleError(
+  notifications: NotificationsService,
+  err: Error
+): Rx.Observable<JobStatusBuckets> {
+  notifications.toasts.addDanger(
+    getGeneralErrorToast(
+      i18n.translate('xpack.reporting.publicNotifier.pollingErrorMessage', {
+        defaultMessage: 'Reporting notifier error!',
+      }),
+      err
+    )
+  );
+  window.console.error(err);
+  return Rx.of({ completed: [], failed: [] });
+}
+
+export class ReportingPublicPlugin implements Plugin<any, any> {
+  private readonly stop$ = new Rx.ReplaySubject(1);
+
+  // FIXME: License checking: only active, non-expired licenses allowed
+  // Depends on https://github.com/elastic/kibana/pull/44922
+  constructor(initializerContext: PluginInitializerContext) {}
+
+  public setup(core: CoreSetup) {}
+
+  // FIXME: only perform these actions for authenticated routes
```

Comment:
Follow up note: this can be possibly be achieved better by registering the job completion notifier as a component of something global for authenticated screens in Kibana, such as the nav header. This is what the Newsfeed Service does, and it is effectively prevented from fetching for news on the Login page due to there being no Nav Header on the Login page.